content-based-message-routing/README.md

```diff
@@ -0,0 +1,494 @@
+
+Use `/healthcare` as the service path (or the context) of the service which is attached to the listener listening on port `port`. Define an HTTP resource that allows the `POST` operation on resource path `/categories/{category}/reserve`, where `category` (corresponding to the specialization) is a path parameter. Use `ReservationRequest` as a parameter indicating that the resource expects a JSON object corresponding to `ReservationRequest` as the payload. Use `ReservationResponse|http:NotFound|http:BadRequest|http:InternalServerError` as the return type to indicate that the response will have a JSON payload corresponding to `ReservationResponse` on success or the response will be a "NotFound", "BadRequest" or "InternalServerError" response on error.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function post categories/[string category]/reserve(ReservationRequest payload) 
+            returns ReservationResponse|http:NotFound|http:BadRequest|http:InternalServerError {
+        
+    }
+```
+
+8. Implement the logic.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function post categories/[string category]/reserve(ReservationRequest payload)
+            returns ReservationResponse|http:NotFound|http:BadRequest|http:InternalServerError {
+        ReservationRequest {hospital_id, patient, ...reservationRequest} = payload;
+        http:Client hospitalEP;
+        match hospital_id {
+            GRANDOAKS => {
+                log:printInfo("Routed to Grand Oak Community Hospital");
+                hospitalEP = grandOakEP;
+            }
+            CLEMENCY => {
+                log:printInfo("Routed to Clemency Medical Center");
+                hospitalEP = clemencyEP;
+            }
+            _ => {
+                log:printInfo("Routed to Pine Valley Community Hospital");
+                hospitalEP = pineValleyEP;
+            }
+        }
+
+        ReservationResponse|http:ClientError resp = hospitalEP->/[category]/reserve.post({
+            patient,
+            ...reservationRequest
+        });
+
+        if resp is ReservationResponse {
+            log:printDebug("Reservation request successful",
+                            name = patient.name,
+                            appointmentNumber = resp.appointmentNumber);
+            return resp;
+        }
+
+        log:printError("Reservation request failed", resp);
+        if resp is http:ClientRequestError {
+            return <http:NotFound> {body: "Unknown hospital, doctor or category"};
+        }
+
+        return <http:InternalServerError> {body: resp.message()};
+    }
+}
+```
+
+- Define `hospitalEP`. Later, this variable is assigned to the relevant client based on the `hospital_id` value.
```

Comment:
Add new lines between content and code blocks.

Reply:
If it is emphasized, it is usually a name, so needs to be a completed sentence. Comment re: assigning variable vs assigning to variable still aplpicable.

```suggestion
- Define the `hospitalEP` variable. Later, the relevant client is assigned to this variable based on the `hospital_id` value.
```

Reply:
Good point. Thanks